Automattic/jetpack
# RNA: In-Place Connection

#### Changes proposed in this Pull Request:
* RNA directory and testing environment.
* The Jetpack's AuthIframe component is partially extracted into the RNA component `InPlaceConnection`.
* Jetpack build and GUI testing script are adjusted to resolve `@automattic/jetpack-connection` to the RNA directory.

#### Jetpack product discussion
p1HpG7-bmK-p2
Partially extracted from the larger PR #18784.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Build Jetpack using `yarn build`, confirm that it builds smoothly.
2. Go to `projects/plugins/jetpack/` and run `yarn test-gui`. Confirm that all tests pass.
3. Go to `projects/js-packages/connection` and run `yarn test`. Confirm that all 5 tests pass.
4. Connect Jetpack, create another WP admin account, and login using it.
5. Go to Jetpack Dashboard and click "Link to WordPress.com" button in the top banner. Confirm that the "Connect your WordPress.com account" iframe shows up.
6. Finish the secondary user connection process. Confirm that it went well.
